src/components/CourseCard/EmailSettingsModal.jsx

```diff
@@ -26,21 +28,12 @@ class EmailSettingsModal extends Component {
 
   handleSaveButtonClick = () => {
     const { hasEmailsEnabled } = this.state;
-    // TODO: Make API POST request to set the new email settings.
+    const { courseRunId, updateEmailSettings } = this.props;
+
     this.setState({
       isSubmitting: true,
     }, () => {
-      /* This callback function simulates an API call to save the
-       * new email settings. There are lines below to either
-       * resolve (200 response) or reject the promise (error response). To
-       * see what happens when the promise is rejected, switch the comments.
-       */
-
-      // eslint-disable-next-line no-unused-vars
-      new Promise((resolve, reject) => {
-        setTimeout(resolve, 2000);
-        // setTimeout(() => reject(new Error('Network Error')), 2000);
-      })
+      updateEmailSettings(courseRunId, hasEmailsEnabled)
```

Comment:
I think we should start moving away from promise chains (`.then`/`.catch`) and instead use async/await. Here's a nice article about why and has some examples: https://blog.pusher.com/promises-async-await/

Simple change that would end up looking something like:
```jsx
this.setState({ isSubmitting: true }, () => {
  try {
    await updateEmailSettings(courseRunId, hasEmailsEnabled);
    this.props.onClose(hasEmailsEnabled);
  } catch (error) {
    this.setState({
      isSubmitting: false,
      error,
    });
  }
});
```

Reply:
alright. can do

Reply:
After wrestling with tests (which turned into a great learning experience), I'd like to opt to do this refactor at a later time to keep the ball rolling on this one!

Reply:
Operating under the impression that @christopappas plans on making a second go at this.